nnvm/tests/python/frontend/tensorflow/test_forward.py

```diff
@@ -787,11 +810,12 @@ def _get_sample(data, state):
                         tvm.nd.array(in_state_h.astype("float32")))
             model.set_input(**params)
             model.run()
-            tvm_output = model.get_output(0, tvm.nd.empty(out_sample_shape,
+            tvm_output = model.get_output(2, tvm.nd.empty(out_sample_shape,
```

Comment:
What are the outputs at 0, 1 here ??

cc @joyalbin  Can you confirm this change ?

Reply:
sym.list_output_names() gives following outputs before these changes were done:
**['Model/Softmax_output', 'concatenate4_output']**

After these changes for multiple outputs, first two outputs are the new outputs that were not added before.
**['__mul_symbol__0_output', '__mul_symbol__1_output', 'Model/Softmax_output', 'concatenate4_output']**

Reply:
Declaring unused nodes this way as output nodes may lead to unwanted computation.

In tensorflow there is no concept of inferring output nodes, instead the intended output node is given by user. I think we could follow the same approach here too. 

Suggest to leave existing frontend as it is. Anyway TVM runtime supports ```get_debug_output``` to specify execution up to required node.

If required we could extend frontend to take output node info to compile a part of the input model.

Reply:
It makes sense that we don't all end nodes to the list of graph outputs to avoid unnecessary computations.

I will add an option for user to provide list of outputs using **from_tensorflow()**. However, by default we should still add all the end nodes to the list of graph outputs. User can still chose the output using **get_debug_output()** to reduce time of execution.

I think we should have both the changes in the frontend: (1) User to provide node(s) as output (2) When user does not provide, then add all end nodes to list of outputs by default. What do you think?

Reply:
Adding all nodes is a problem as it changes output indexes. Cases like nodes which are valid only in training may lead unexpected output nodes.

Hence suggest not to auto infer output nodes, but use last node as output. For a subgraph compilation we could give an option to user which is similar to tensorflow.

Reply:
In the latest commit, I have removed addition of multiple outputs. Now, user can provide list of outputs to from_tensorflow() API. Please have a look.